specs/DesignDiscussions/InputLabels.md

```diff
@@ -0,0 +1,279 @@
+	</TextBox.Header>
+</TextBox>
+```
+![img](assets/InputLabels/XamlHeaderFormat.png)
+
+Other control types such as Images can also be used as TextBox labels.
+
+#### AutomationProperties
+ Xaml also has an AutomationProperties.LabeledBy attached property for accessibility purposes. This property is used by all elements (not just inputs). If you use it to mark a proceeding text block as the label, it is read the same as a header by narrator, but does not have the click to select behavior of header. 
+
+ ```xml
+<TextBlock x:Name="inputLabel">This is a text block label</TextBlock>
+<TextBox AutomationProperties.LabeledBy="{Binding ElementName=inputLabel}"/> 
+```
+ ![img](assets/InputLabels/XamlLabeledBy.png)
+
+### HTML
+#### Label
+HTML has label tag which can be used to label inputs. The input can either be placed inside the label, or referenced with “for”:
+
+``` HTML
+  <label>
+  Please enter your name:
+  <input id="nameInput">
+  </label>
+```
+
+Or
+``` HTML
+  <label for="nameInput"> Please enter your name:</label>
+  <input id="nameInput">
+```
+ 
+This results in the label being associated with the input both for interactivity and accessibility.
+
+#### Aria
+ HTML also has aria-label and aria-labelledby. These properties are used for accessibility purposes. As in the Xaml AutomationProperties.LabeledBy property, this labels the input for accessibility purposes, but does not change the interactivity model: clicking on the label does not select the input.
+
+## Adaptive Card Schema Options for Semantic Labels
+Below are two possible options for schemas associating labels and inputs. I believe option 1 is the better option, although option 2 allows the additional scenario of having a label separate from the input.
+
+### Option 1 – Label property on inputs
+
+One option would be to make the label text a property of the input itself. This is analogous to the XAML header property discussed above.
+
+``` json
+{
+	"type": "AdaptiveCard",
+	"version": "1.0",
+	"body": [
+		{
+			"type": "Input.Text",
+			"id": "nameInput",
+			"label": "Please enter your name:",
+			"placeholder": "Last, First"
+		}
+	],
+	"actions": [
+		{
+			"type": "Action.Submit",
+			"title": "OK"
+		}
+	]
+}
+```
+This has the benefit of being simple and concise, but doesn’t immediately allow for formatting on the input label. A solution to that would be to add support for TextBlock and/or RichTextBlock labels to add formatting support. This is similar to the approach in Xaml:
+
+```json
+{
+	"type": "Input.Text",
+    "id": "nameInput",
+    "label": 
+    {
+        "type": "TextBlock",
+        "text": "Please enter your name:",
+        "weight": "Bolder",
+        "color": "Good"
+    },
+    "placeholder": "Last, First"
+}
+```
+
+We should also consider which item types are valid as labels. Probably just TextBlock and RichTextBlock make sense, but as Xaml supports other types we should consider whether we want to as well.
```

Comment:
Discussion in room: Probably fine with text types only, but will follow up with customers once we have a prototype.